There is some problem, repro:

```bash
$ git clone https://github.com/ClickHouse/ClickHouse.git
$ cd ClickHouse/
# add '--remote' to contrib/update-submodules.sh
$ ./contrib/update-submodules.sh
# cmake -S . -B build
```

fails with

```
CMake Error at contrib/llvm-project/llvm/CMakeLists.txt:82 (message):
  Requested CMAKE_CXX_STANDARD=14 which is less than the required 17.
```

When one does not add `--remote` and runs the repro again, one will see that the submodule in `contrib/llvm-project/` is on a different commit.

--- a/contrib/update-submodules.sh
+++ b/contrib/update-submodules.sh
@@ -9,4 +9,8 @@ cd $GIT_DIR
 contrib/sparse-checkout/setup-sparse-checkout.sh
 git submodule init
 git submodule sync
-git config --file .gitmodules --get-regexp .*path | sed 's/[^ ]* //' | xargs -I _ --max-procs 64 git submodule update --depth=1 --single-branch _
+# NOTE: do not use --remote for `git submodule update`[1] command, since the submodule references to the specific commit SHA1 in the subproject.
+#       It may cause unexpected behavior. Instead you need to commit a new SHA1 for a submodule.
+#
+#       [1] - https://git-scm.com/book/en/v2/Git-Tools-Submodules
+git config --file .gitmodules --get-regexp '.*path' | sed 's/[^ ]* //' | xargs -I _ --max-procs 64 git submodule update --depth=1 --single-branch _

